beard/clustering/wrappers.py

```diff
@@ -79,17 +79,24 @@ def __init__(self, method="single", affinity="euclidean",
             The statistics upon which non-singleton i is thresholded.
             See scipy.cluster.hierarchy.fcluster for further details.
 
+         :param scoring_data: string or None
+            The form of the input data
+            - "raw": Array of samples;
+            - "affinity": Affinity matrix.
+
         :param scoring: callable
             The scoring function to maximize in order to estimate the best
-            threshold. There are 4 possibles cases based on data availability:
-            - ground_truth and affinity: scoring(X, labels_true, labels_pred)
-            - ground_truth but not affinity: scoring(labels_true, labels_pred)
-            - affinity but not ground_truth: scoring(X, labels_pred)
-            - none: scoring(labels_pred).
-
-        :param affinity_score: boolean
-            A flag that must be True if the scoring function requires the
-            affinity as an input. False otherwise.
+            threshold. There are 6 possible cases based on the availability of
+            ground truth data passed in y and the scoring_data parameter value:
+            (Notations definition: Xr means X when it contains the raw samples;
+                Xa means X when it is the affinity matrix of the samples.)
+            - ground_truth and Xr: scoring(Xr, label_true, label_pred);
```

Comment:
Rephrase it as follows:

- Ground truth data and `scoring_data == "raw"`: 
- No ground truth data and `scoring_data == "raw"`: 
- Ground truth data and `scoring_data == "affinity"`: 
- No ground truth data and `scoring_data == "affinity"`: 
- Ground truth data and `scoring_data is None`: 
- No ground truth data and `scoring_data is None`: